minix/lib/libacpi/acpi.c

```diff
@@ -0,0 +1,95 @@
+#define _SYSTEM
+
+#include <errno.h>
+#include <string.h>
+
+#include <minix/acpi.h>
+#include <minix/com.h>
+#include <minix/ds.h>
+#include <minix/ipc.h>
+#include <minix/log.h>
+#include <minix/sysutil.h>
+
+#include "acpi.h"
+
+static struct log log =
+{ .name = "libacpi", .log_level = LEVEL_TRACE, .log_func = default_log };
+
+static endpoint_t acpi_ep = NONE;
+
+int
+do_init()
+{
+	int res;
+	res = ds_retrieve_label_endpt("acpi", &acpi_ep);
+	log_info(&log, "resolved acpi to endpoint: %d\n", acpi_ep);
+	return res;
+}
+
+int
+acpi_init()
+{
+	return do_init();
+}
+
+/*===========================================================================*
+ *				IRQ handling				     *
+ *===========================================================================*/
+int
+acpi_get_irq(unsigned bus, unsigned dev, unsigned pin)
+{
+	int err;
+	message m;
+
+	if(acpi_ep == NONE)
```

Comment:
When writing new code, please follow the NetBSD KNF style guide at least for the more important things:

  http://git.minix3.org/index.cgi?p=minix.git;a=blob;f=share/misc/style;h=5eedbc1532d30fbd29ab4946571b3cf956cf25f3;hb=HEAD

Most importantly, there should be a space between 'if' and the condition, and for non-function blocks the opening brace should be on the same line basically.